app/services/sources/sources.ts

```diff
@@ -314,7 +299,20 @@ export class SourcesService extends StatefulService<ISourcesState> implements IS
       activeItems.forEach(activeItem => {
         sourcesNames.push(activeItem.sourceId);
       });
-      ipcRenderer.send('requestSourceAttributes', sourcesNames);
+
+      const sizes: obs.ISourceSize[] = obs.getSourcesSize(sourcesNames);
```

Comment:
Is this synchronous?  The main reason we did the the weird thing with events to the main process is because we do this every second in the background and it's very disruptive to do it synchronously.

Reply:
It would previously block the main process. Now it blocks the renderer for ~20µs or so. I have a branch for obs-studio-node which actually makes the function asynchronous, if we want to go for that.

Reply:
Yeah I was talking to Eddy, and I think eventually it would be awesome if the IPC client supported calling functions both sync and async.  Then in the frontend we can make that decision whether we want to block or not.

For now I think it's fine as long as we test this with a very large scene collection (50 - 100 sources) on a moderate speed computer and see how it compares to our current live release.